Sources/MUXSDKStatsInternal/MUXSDKVideoData.swift

```diff
@@ -0,0 +1,214 @@
+import AVFoundation
+import MuxCore
+
+extension MUXSDKVideoData {
+    func updateWithPresentationSize(_ presentationSize: CGSize) {
+        videoSourceHeight = presentationSize.height as NSNumber
+        videoSourceWidth = presentationSize.width as NSNumber
+    }
+
+    @available(iOS 15, tvOS 15, *)
+    func updateWithAssetVariant(_ assetVariant: AVAssetVariant) {
+        if let averageBitRate = assetVariant.averageBitRate {
+            videoSourceAdvertisedBitrate = averageBitRate as NSNumber
+        }
+
+        assetVariant.videoAttributes.map(updateWithVideoAttributes(_:))
+    }
+
+    @available(iOS 15, tvOS 15, *)
+    func updateWithVideoAttributes(_ videoAttributes: AVAssetVariant.VideoAttributes) {
+        if let nominalFrameRate = videoAttributes.nominalFrameRate {
+            videoSourceAdvertisedFrameRate = nominalFrameRate as NSNumber
+        }
+
+        if videoAttributes.presentationSize != .zero {
+            updateWithPresentationSize(videoAttributes.presentationSize)
+        }
+
+        // TODO: videoAttributes.codecTypes
+    }
+
+    @available(iOS 15, tvOS 15, *)
+    func updateWithRenditionInfo(track: AVAssetTrack, on playerItem: AVPlayerItem) async {
+        // This is less exact and only used after all other variant matching options are exhausted
+        // Still obtain it immediately in case another variant switch happens
```

Comment:
I'm not sure if I understand this comment. Is it about `indicatedBitratesInAccessLog` or is it about the overall usage of this method?

I see that we use `updateWithRenditionInfo` for getting our initial rendition, and when we don't have AVMetrics available, which seems like an interpretation of the comment? But I also see that we only use `bitratesFromAccessLog` conditionally, which also seems like an interpretation of the comment 😅 .